model/vm.rb

```diff
@@ -197,6 +197,7 @@ def params_json(swap_size_bytes)
     topo = cloud_hypervisor_cpu_topology
 
     project_public_keys = project.get_ff_vm_public_ssh_keys || []
+    public_keys = public_key.split("\n").map(&:strip).reject { _1.nil? || _1.empty? }
```

Comment:
```suggestion
    public_keys = public_key.split("\n").each(&:strip!)
    public_keys.reject!(&:empty?)
    public_keys.concat(project_public_keys)
```

I don't believe `String#strip` can return `nil`.  Since `String#split` returns a newly allocated array of newly allocated strings, `.map(&:strip)` => `.each(&:strip!)` has the same behavior, but does not allocate a new array and new strings.  Likewise, `reject` -> `reject!` saves an array allocation.  And using `concat` instead of `+` also saves an array allocation.

Reply:
I applied changes